### PR TITLE
4.x cache cleanup

### PR DESCRIPTION
**Pull request by ADmad**
Increased code coverage a bit.

### CODE REVIEW

**Comment by ADmad** [comment]
I noticed that none of the engine classes use the specific methods provided by their corresponding engines to get / set multiple cache keys, why is that? All cache use the implementation in `CacheEngine::setMultiple()` and `CacheEngine::getMultiple()` which simply loops over the methods for getting/setting single key.

File: src/Cache/CacheEngine.php [hunk 1]
```diff
@@ -307,10 +306,7 @@ public function add(string $key, $value): bool
      * @param string $group name of the group to be cleared
      * @return bool
      */
-    public function clearGroup(string $group): bool
-    {
-        return false;
-    }
+    abstract public function clearGroup(string $group): bool;
```
Comment (ADmad) [inline src/Cache/CacheEngine.php]: All engines have their own implementation for this method so this implementation was never used.

File: tests/TestCase/Cache/Engine/FileEngineTest.php [hunk 2]
```diff
@@ -189,11 +189,13 @@ public function testSetWithTtl()
         $this->assertTrue($engine->set('default_ttl', $data));
         $this->assertTrue($engine->set('int_ttl', $data, 1));
         $this->assertTrue($engine->set('interval_ttl', $data, new DateInterval('PT1S')));
+        $this->assertTrue($engine->setMultiple(['multi' => $data], 1));
```
Comment (ADmad) [inline tests/TestCase/Cache/Engine/FileEngineTest.php]: `CacheEngine::setMultiple()` with non-null `$ttl` didn't have any test earlier.

**Comment by markstory** [comment]
> All classes use the implementation in CacheEngine::setMultiple() and CacheEngine::getMultiple() which simply loops over the methods for getting/setting single key.

Probably an oversight. We could fix that though and use multi-set where available.

**Comment by ADmad** [comment]
Okay, merging this. We can deal with the multi-methods in a separate PR.

**Pull request merged**